File: mobile/app/api/types/createDogRequest.ts
/**
 * 
 * @export
 * @interface CreateDogRequest
 */

export interface CreateDogRequest {
    /**
     * 
     * @type {string}
     * @memberof CreateDogRequest
     */
    'name': string;
    /**
     * 
     * @type {number}
     * @memberof CreateDogRequest
     */
    'age'?: number;
    /**
     * 
     * @type {string}
     * @memberof CreateDogRequest
     */
    'size': string;
    /**
     * 
     * @type {string}
     * @memberof CreateDogRequest
     */
    'energy': string;
    /**
     * 
     * @type {string}
     * @memberof CreateDogRequest
     */
    'breed'?: string;
    /**
     * 
     * @type {string}
     * @memberof CreateDogRequest
     */
    'bio'?: string;
    /**
     * 
     * @type {Array<string>}
     * @memberof CreateDogRequest
     */
    'photos'?: string[];
}

File: api/src/main/java/com/dkowalczyk/psinder_app/dto/CreateDogRequest.java
package com.dkowalczyk.psinder_app.dto;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CreateDogRequest {
    @NotBlank(message = "Dog name is required")
    private String name;
    
    @Min(value = 0, message = "Dog age cannot be negative")
    private int age;
    
    @NotBlank(message = "Dog size is required (SMALL, MEDIUM, LARGE)")
    private String size;
    
    @NotBlank(message = "Dog energy is required (SMALL, MEDIUM, HIGH, VERY_HIGH)")
    private String energy;
    private String breed;
    private String bio;
    private List<String> photos;
}
